resources/js/confirm-delete.js

```diff
@@ -1,10 +1,45 @@
+// Show confirm dialog before submitting any spoofed DELETE method form
 document.body.addEventListener("submit", function(event) {
-  if (event.target.querySelector('input[name="_method"][value="delete"], input[name="_method"][value="DELETE"]')) {
+  let method;
+  let deleteButton;
+
+  // Take last spoofed method using FormData if available
+  if (FormData instanceof Object) {
+    let formData = new FormData(event.target);
+    if (typeof formData.getAll == "function") {
+      method = (
+        formData.getAll("_method").pop() || event.target.method
+      ).toUpperCase();
+    }
+  }
+
+  [...event.target.elements].forEach(function(input) {
+    // Take first spoofed method if FormData was not available
+    if (!method && input.name == "_method") {
+      method = input.value.toUpperCase();
+    }
+    // Take first submit button
+    if (!deleteButton && input.type == "submit") {
+      deleteButton = input;
+    }
+    // Take last DELETE button over first button
+    if (
+      input.type == "submit" &&
+      input.name == "_method" &&
+      input.value.toUpperCase() == "DELETE"
+    ) {
+      deleteButton = input;
+    }
```

Comment:
@bjuppa Just some thoughts about the code above, sorry if I have gotten this absolutely wrong! 

First off, in what cases do we want to use the first submit button? 

To make it a bit more general (i.e. if we would add more types of delete actions?!) we could perhaps just change the deleteButton variable to be called methodButton, and not consider the action here, and do something like:

  if (
      input.type == "submit" &&
      input.name == "_method" &&
    ) {
      methodButton = input;
    }

And otherwise not set it?

Reply:
I wanted this script to only care about DELETE methods so there's no use for a button reference in the script if it's not *the* delete button. So basically if the script can't figure out the right button, but the method is cleary set to DELETE, then the script will at least display the default "Delete item" text, instead of some other text from another button.

Reply:
Sorry, I didn't answer your first question!

In most cases where the form is a dedicated delete form, the submit button won't have any `_method` set, the method will just be in the hidden input. So the first button is the one that'd be activated by pressing Enter within the form.

...but if there's a dedicated delete button later down the form, we'd rather use the message from that button to display in the popup (if the submit indicates a DELETE method), even though it might not have been the actual button clicked, that button should hold the most relevant message for the popup!

Reply:
One could say the script does two things in parallell:
The first is to determine whether or not the current submission is with a DELETE method.
The other is to determine what message to display in a delete confirmation popup, should the first condition be true!